docs/openshift/deploy.md

```diff
@@ -117,3 +117,53 @@ Run `acs-engine generate examples/openshift.json`
 After couple of minutes, your OpenShift web console should be accessible at `https://${dnsprefix}.${location}.cloudapp.azure.com:8443/`. You can log in with `clusterUsername` and `clusterPassword` values you set in the `openshift.json` file or with your Azure account if you've used AAD integration.
 
 For next steps, see [getting started documentation](https://docs.openshift.org/latest/getting_started/index.html) on OpenShift website.
+
+## Custom VNET
+
+ACS Engine supports deploying into an existing VNET. Operators must specify the ARM path/id of Subnets for the `masterProfile` and  any `agentPoolProfiles`, as well as the first IP address to use for IP static IP allocation in `firstConsecutiveStaticIP`. Additionally, to prevent source address NAT'ing within the VNET, we assign to the `vnetCidr` property in `masterProfile` the CIDR block that represents the usable address space in the existing VNET.
+
+To create a vnet and a subnet, for example:
+
+```bash
+az network vnet create -g $RESOURCE_GROUP -n $VNET_NAME --address-prefixes 10.239.0.0/16 --subnet-name $SUBNET_NAME --subnet-prefix 10.239.0.0/24
+```
+To get the `vnetSubnetId`:
+
+```bash
+az network vnet subnet show -n $SUBNET_NAME -g $RESOURCE_GROUP --vnet-name $VNET_NAME | jq -r .id
+```
+
+Edit the [OpenShift with custom vnet cluster definition](/examples/vnet/openshift-vnet.json) and fill out the required values (every value with empty default `""` must be filled in).
```

Comment:
@jim-minter @kargakis @kwoodson we need to makes sure to keep this example up to date now too :smile:

Reply:
We already use the example in the e2e tests so it needs to be updated anyway.

https://github.com/Azure/acs-engine/blob/bfc9c7eea789970aa87287183d7123ee5bc69ad9/.circleci/config.yml#L369

Reply:
Oops, sorry this is a different example. I would expect the e2e test that validates custom vnets would use the new example.

Reply:
follow-up https://github.com/Azure/acs-engine/issues/2984 to implement unit test to keep openshift examples in sync

Reply:
follow-up https://github.com/Azure/acs-engine/issues/2985 to implement custom vnet e2e test

Reply:
https://github.com/Azure/acs-engine/pull/2986 for #2984